### DISCUSSION TITLE
How to use proxy for intranet access?

### DISCUSSION THREAD

**Discussion by mike-668** [XTLS/Xray-core#3723] [General]
Hello everyone,
The client router's intranet IP is 192.168.8.0/24, and the server router's intranet IP is 172.18.1.0/24. How can I set up Xray's proxy rules so that the intranet IPs on both ends can access each other through the proxy?

**Comment by kingwilliam** [comment]
you mean your network like this?
```
                          /---------------\                         /---------------\
----[ 192.168.8.0/24 ]----| client router |----[ 172.18.1.0/24 ]----| server router |-----
                          \---------------/                         \---------------/
```

**Reply by mike-668** [reply]
Yes, the Client router has no public IP, the Server router has a fixed public IP, and their private IPs are as you show.

**Reply by kingwilliam** [reply]
if you want both subnet can access each other, why not change client router to soft router (like pfsense or opnsense)

**Reply by mike-668** [reply]
Through VPN, it is easy to achieve internal network access between two routes. Due to firewall blocking, ordinary VPN cannot be established. When the X-ray proxy software establishes a connection, a virtual tunnel interface is not created, which makes it difficult to implement this theme; I originally intended to pass through“ https://github.com/XTLS/Xray-core/discussions/3725 To achieve it, but without success

**Reply by kingwilliam** [reply]
if client router use soft router, access between two subnet only need ip routing no need vpn, 

wireguard over xray, I run this config smoothly, but my environment not same as your environment, so i cannot give you any suggestion to trouble shoot you config

----------------

in normal status(NAT client router), 192 subnet can access 172 subnet and internet
in normal status(NAT client router), 172 subnet CANNOT access 192 subnet, but can use internet

if you want to use xray form sock5/http proxy to serve 172 subnet to access 192 subnet, 
you can stand alone 1pc at 192 subnet(192.168.0.111) port 1080 and 
create 1080 port forward at client router to 192.168.0.111
(but,  you must have admin rights at client router)

----------------

OR can you DETAILS describe your idea full picture, what problem exists on your network.

AND do you have admin rights on your client router? I means not only config rights, I meaning change the hardware
